tools/cpp/windows_cc_configure.bzl

```diff
@@ -31,20 +31,24 @@ def _auto_configure_warning_maybe(repository_ctx, msg):
         auto_configure_warning(msg)
 
 def _get_escaped_windows_msys_crosstool_content(repository_ctx, use_mingw = False):
-    """Return the content of msys crosstool which is still the default CROSSTOOL on Windows."""
-    bazel_sh = get_env_var(repository_ctx, "BAZEL_SH").replace("\\", "/").lower()
+    """Return the content of msys crosstool."""
+    bazel_sh = get_env_var(repository_ctx, "BAZEL_SH", "", False).replace("\\", "/").lower()
     tokens = bazel_sh.rsplit("/", 1)
-    prefix = "mingw64" if use_mingw else "usr"
-    msys_root = None
+    msys_root = ""
     if tokens[0].endswith("/usr/bin"):
         msys_root = tokens[0][:len(tokens[0]) - len("usr/bin")]
     elif tokens[0].endswith("/bin"):
         msys_root = tokens[0][:len(tokens[0]) - len("bin")]
-    if not msys_root:
-        auto_configure_fail(
-            "Could not determine MSYS/Cygwin root from BAZEL_SH (%s)" % bazel_sh,
-        )
-    escaped_msys_root = escape_string(msys_root)
+    prefix = "mingw64" if use_mingw else "usr"
+    tool_path_prefix = escape_string(msys_root) + prefix
+    tool_path = {}
+
+    for tool in ["ar", "compat-ld", "cpp", "dwp", "gcc", "gcov", "ld", "nm", "objcopy", "objdump", "strip"]:
+        if not msys_root:
```

Comment:
Flipping the branches to get `if [condition] then [action] else [alternative]` is easier to read than `if not [condition] then [alternative] else [action]`.

Reply:
Indeed!